scripts/update-values-file-image-tag.sh

```diff
@@ -1,26 +1,45 @@
 #!/bin/bash
 
-# Note:
+# NOTE:
+# The whole script will be removed once we have RenovateBot setup in place.
+
+# NOTE:
 # The "yq" tool is not unsable because of this bug:
 # https://github.com/mikefarah/yq/issues/515
 # When the bug is fixed we can use:
-# 	yq -i '.global.image.tag = env(GLOBAL_IMAGE_TAG)' charts/camunda-platform/values.yaml
-# 	yq -i '.optimize.image.tag = env(OPTIMIZE_IMAGE_TAG)' charts/camunda-platform/values.yaml
+# 	yq -i '.global.image.tag = env(IMAGE_TAG_GLOBAL)' charts/camunda-platform/values.yaml
 
-if [[ -z "${GLOBAL_IMAGE_TAG}" && -z "${OPTIMIZE_IMAGE_TAG}" ]]; then \
-    echo '[ERROR] One or both of the following vars should be defined:'
-		echo -e "  - GLOBAL_IMAGE_TAG\n  - OPTIMIZE_IMAGE_TAG"
-    exit 1
-fi
+declare -A componenets_tags
+declare -A componenets_values
 
-if [[ -n "${GLOBAL_IMAGE_TAG}" ]]; then
-    sed -ri "s/(\s+)tag:.+# (global.image.tag)/\1tag: ${GLOBAL_IMAGE_TAG}  # \2/g" \
-        charts/camunda-platform/values.yaml
-    echo "Updated global.image.tag=${GLOBAL_IMAGE_TAG}"
-fi
+componenets_tags+=(
+  [GLOBAL]=global.image.tag
+  [OPTIMIZE]=optimize.image.tag
+  [WEBMODELER]=webModeler.image.tag
+  [CONNECTORS]=connectors.image.tag
+)
 
-if [[ -n "${OPTIMIZE_IMAGE_TAG}" ]]; then
-    sed -ri "s/(\s+)tag:.+# (optimize.image.tag)/\1tag: ${OPTIMIZE_IMAGE_TAG}  # \2/g" \
-        charts/camunda-platform/values.yaml
-    echo "Updated optimize.image.tag=${OPTIMIZE_IMAGE_TAG}"
+componenets_values+=(
+  [GLOBAL]=${IMAGE_TAG_GLOBAL}
+  [OPTIMIZE]=${IMAGE_TAG_OPTIMIZE}
+  [WEBMODELER]=${IMAGE_TAG_WEBMODELER}
+  [CONNECTORS]=${IMAGE_TAG_CONNECTORS}
+)
+
+# Ensure that there is at least 1 env var exported to avoid silent errors.
+if [[ $(echo "${componenets_values[@]}" | tr -d '[:space:]') == "" ]]; then
+    echo '[ERROR] One of the following vars should be defined:'
+    printf -- '- IMAGE_TAG_%s\n' "${!componenets_values[@]}"
+    exit 1
 fi
+
+# Update values.yaml file with the exported vars.
+for key in "${!componenets_tags[@]}"; do
```

Comment:
Is this typo consistent?

Reply:
that's what I'm talking about 😂
it's always good to have another pair of eyes :grin: